core/src/HIP/Kokkos_HIP_Locks.hpp

```diff
@@ -138,20 +135,6 @@ inline static void ensure_hip_lock_arrays_on_device() {
 }
 #endif
 
-#else
-
-#ifdef KOKKOS_ENABLE_HIP_RELOCATABLE_DEVICE_CODE
-inline void ensure_hip_lock_arrays_on_device() {}
-#else
-// Still Need copy_hip_lock_arrays for team scratch etc.
-inline static void ensure_hip_lock_arrays_on_device() {
-  copy_hip_lock_arrays_to_device();
-  desul::ensure_hip_lock_arrays_on_device();
```

Comment:
This looks like you deleted the wrong branch. We should keep the one that has
```
desul::ensure_hip_lock_arrays_on_device();
```

Reply:
Yup you are right

Reply:
Fixed